Functional Programming/Problem2/Problem2.js

```diff
@@ -0,0 +1,29 @@
+(function () {
+  'use strict';
+
+  function curry(fn) {
+    var length = fn.length;
+
+    function wrap(prev) {
+      return function(arg) {
+        var args = prev.concat(arg);
+        if (args.length < length) {
+          return wrap(args);
+        } else {
+          return fn.apply(this, args);
+        }
+      }
+    }
+
+    return wrap([]);
```

Comment:
Wrap function could be omitted. Put args in closure as well as length. Then you can return a single function instead of new one for each call.